File: csharp-netstandard/src/Cloud.Governance.Client/Model/SiteSharingStatus.cs


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = Cloud.Governance.Client.Client.OpenAPIDateConverter;

namespace Cloud.Governance.Client.Model
{
    
    [JsonConverter(typeof(StringEnumConverter))]
    
    public enum SiteSharingStatus
    {
        [EnumMember(Value = "0")]
        None = 0,

        [EnumMember(Value = "1")]
        AnyOne = 1,

        [EnumMember(Value = "2")]
        NewAndExistingGuest = 2,

        [EnumMember(Value = "3")]
        ExistingGuestOnly = 3,

        [EnumMember(Value = "4")]
        Disabled = 4

    }

}

File: typescript/src/models/SiteSharingStatus.ts
/* tslint:disable */
/* eslint-disable */
/*
 * Cloud Governance Api
 *
 * Contact: support@avepoint.com
 */

/**
 * 
 * @export
 * @enum {string}
 */
export enum SiteSharingStatus {
    None = 0,
    AnyOne = 1,
    NewAndExistingGuest = 2,
    ExistingGuestOnly = 3,
    Disabled = 4
}

export function SiteSharingStatusFromJSON(json: any): SiteSharingStatus {
    return SiteSharingStatusFromJSONTyped(json, false);
}

export function SiteSharingStatusFromJSONTyped(json: any, ignoreDiscriminator: boolean): SiteSharingStatus {
    return json as SiteSharingStatus;
}

export function SiteSharingStatusToJSON(value?: SiteSharingStatus | null): any {
    return value as any;
}

